dashboard/app/controllers/api_controller.rb

```diff
@@ -342,6 +342,41 @@ def section_level_progress
     }
   end
 
+  # GET /api/teacher_panel_progress/:section_id
+  # Get complete details of a particular section for the teacher panel progress
+  def teacher_panel_progress
+    section = load_section
+    script = load_script(section)
+
+    if params[:level_id]
+      script_level = script.script_levels.find do |sl|
+        sl.level_ids.include?(params[:level_id].to_i)
+      end
+      return head :bad_request if script_level.nil?
```

Comment:
Is it worth returning some kind of message describing the issue?

Reply:
I'm thinking no?  This would be mostly for debugging purposes and it seems like this would be easy enough to diagnose with just the error code?  (famous last words)